_query-dsl/specialized/script-score.md

```diff
@@ -0,0 +1,332 @@
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.2876821,
+    "hits": [
+      {
+        "_index": "testindex1",
+        "_id": "1",
+        "_score": 0.2876821,
+        "_source": {
+          "name": "John Doe",
+          "multiplier": 0.5
+        }
+      }
+    ]
+  }
+}
+```
+
+Now, let's change the document score with a script that calculates the score as the value of the `_score` field multiplied by the value of the `multiplier` field. In the following query, you can access the current relevance score of a document in the `_score` variable and the `multiplier` value as `doc['multiplier'].value`:
+
+```json
+GET testindex1/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match": { 
+            "name": "John" 
+        }
+      },
+      "script": {
+        "source": "_score * doc['multiplier'].value"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In the response, the score for document 1 is half of the original score:
+
+```json
+{
+  "took": 8,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.14384104,
+    "hits": [
+      {
+        "_index": "testindex1",
+        "_id": "1",
+        "_score": 0.14384104,
+        "_source": {
+          "name": "John Doe",
+          "multiplier": 0.5
+        }
+      }
+    ]
+  }
+}
+```
+
+## Parameters
+
+The `script_score` query supports the following top-level parameters.
+
+Parameter | Data type | Description
+:--- | :--- | :---
+`query` | Object | The query to use for search. Required.
+`script` | Object | The script to use to calculate the score of the documents that the `query` returns. Required.
+`min_score` | Float | Excludes documents with a score lower than `min_score` from the results. Optional.
+`boost` | Float | Boosts the documents' scores by the given multiplier. Values less than 1.0 decrease relevance, and values greater than 1.0 increase relevance. Default is 1.0. 
+
+The relevance scores calculated by the `script_score` query cannot be negative. 
+{: .important}
+
+## Customizing score calculation with built-in functions
+
+To customize score calculation, you can use one of the built-in Painless functions. For every function, OpenSearch provides one or more Painless methods you can access in the script score context. You can call the Painless methods listed in the following sections directly, without using a class name or instance name qualifier.
+
+### Saturation
+
+The saturation function calculates saturation as `score = value /(value + pivot)`, where `value` is the field value and `pivot` is chosen so that the score is greater than 0.5 if `value` is greater than `pivot` and less than 0.5 if `value` is less than `pivot`. The score is in the (0, 1) range. To apply a saturation function, call the following Painless method:
```

Comment:
```suggestion
The saturation function calculates saturation as `score = value /(value + pivot)`, where `value` is the field value and `pivot` is chosen so that the score is higher than 0.5 if `value` is higher than `pivot` and lower than 0.5 if `value` is lower than `pivot`. The score is in the (0, 1) range. To apply a saturation function, call the following Painless method:
```